### PR TITLE
feat: Simplify recording player

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We have a lot of props drilling in the recording player which is both unnecessary and hard to read.

## Changes

* Takes advantage of BindLogic to set props once and have that apply across the player components

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/notebooks/Nodes/RecordingNode.tsx [hunk 1]
```diff
@@ -17,7 +16,8 @@ const Component = (props: NodeViewProps): JSX.Element => {
             {...props}
             className={NotebookNodeType.Recording}
             title="Recording"
-            preview={<PlayerMeta {...recordingLogicProps} />}
+            // TODO: Fix "meta" preview
+            // preview={<PlayerMeta {...recordingLogicProps} />}
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/RecordingNode.tsx:20]: We'll fix this in the notebooks sprint

**Review by pauldambra — APPROVED** [review]
:shipit:

File: frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx [hunk 2]
```diff
@@ -7,20 +7,18 @@ import { PlaylistPopover } from './playlist-popover/PlaylistPopover'
...
+export function PlayerMetaLinks(): JSX.Element {
+    const { sessionRecordingId, logicProps } = useValues(sessionRecordingPlayerLogic)
+    const { setPause, deleteRecording } = useActions(sessionRecordingPlayerLogic)
     const { featureFlags } = useValues(featureFlagLogic)
 
     const onShare = (): void => {
         setPause()
         openPlayerShareDialog({
-            seconds: Math.floor((logic.values.currentPlayerTime || 0) / 1000),
+            seconds: Math.floor((sessionRecordingPlayerLogic.values.currentPlayerTime || 0) / 1000),
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx:21]: I'm out of context and can't say what was desired here, but the difference is mostly:
- `const { currentPlayerTime } = useValues(logic)` - we re-render the component every time `a` changes, and mount the logic if not mounted
- `const currentPlayerTime = logic.values.currentPlayerTime` - no re-render every second, only access the variable in the callback when it's called.

Not sure if the above is intentional, but swapping this to `useValues` will likely make things a little bit slower.

[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx:21]: It was exactly for that reason. We only need the value when it is clicked. Generally we try and never include the playertime unless really necessary as it changes a lot

[the remark this answers is not in the document]
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx:21]: Ah, cool! #ThingsILearned thanks both!

**Pull request merged**